openhtf/plugs/user_input.py

```diff
@@ -194,13 +194,15 @@ def respond(self, prompt_id, response):
 
 
 def prompt_for_test_start(
-    message='Enter a DUT ID in order to start the test.', timeout_s=60*60*24):
+    message='Enter a DUT ID in order to start the test.', timeout_s=60*60*24,
+    upper=False):
```

Comment:
I'd prefer a validation hook that's a little more general and you can provide `string.upper` but in the future you may want to validate serials more thoroughly.

Take `validator=None` as an argument and put this below:

```py
if validator is None:
  validator = lambda sn: sn
test.test_record.dut_id = validator(dut_id)
```